kotlin/animate-images-with-image-overlay/src/main/java/com/esri/arcgisruntime/sample/animateimageswithimageoverlay/MainActivity.kt

```diff
@@ -0,0 +1,237 @@
+      }
+    }
+
+    // get the image files from local storage as an unordered list
+    (File(getExternalFilesDir(null).toString() + "/ImageFrames/PacificSouthWest").listFiles())?.let { imageFiles ->
+      // sort the list of image files
+      Arrays.sort(imageFiles)
+      imageFiles.forEach { file ->
+        // create an image with the given path and use it to create an image frame
+        val imageFrame = ImageFrame(file.path, pacificSouthwestEnvelope)
+        imageFrames.add(imageFrame)
+      }
+    }
+
+    // show the options sheet when the floating action button is clicked
+    fab.setOnClickListener {
+      fab.isExpanded = !fab.isExpanded
+    }
+
+    // seek bar controls opacity
+    opacitySeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+      override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+        // convert the seekbar progress (0 - 100) to a float 0.0 - 1.0
+        val opacity: Float = progress.toFloat() / 100
+        sceneView.imageOverlays[0].opacity = opacity
+        currOpacityTextView.text = opacity.toString()
+      }
+
+      override fun onStartTrackingTouch(seekBar: SeekBar?) {}
+      override fun onStopTrackingTouch(seekBar: SeekBar?) {}
+    })
+
+    // spinner controls how many image overlays to display per second
+    fpsSpinner.apply {
+      // create an adapter with fps options
+      adapter = ArrayAdapter(
+        this@MainActivity,
+        android.R.layout.simple_spinner_dropdown_item,
+        arrayOf("60 fps", "30 fps", "15 fps")
+      )
+      // set period based on the fps option selected
+      onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+        override fun onItemSelected(
+          parent: AdapterView<*>?,
+          view: View?,
+          position: Int,
+          id: Long
+        ) {
+          // get the period for the chosen fps
+          period = when (position) {
+            0 -> 17 // 1000ms/17 = 60 fps
+            1 -> 33 // 1000ms/33 = 30 fps
+            2 -> 67 // 1000ms/67 = 15 fps
+            else -> 0
+          }
+          // create a new timer for this period
+          if (isTimerRunning) {
+            timer?.cancel()
+            createNewTimer()
+          }
+        }
+
+        override fun onNothingSelected(parent: AdapterView<*>?) {}
+      }
+      // start at 15 fps
+      setSelection(2)
+    }
+  }
+
+  /**
+   * Create a new image frame from the image at the current index and add it to the image overlay.
+   */
+  private fun addNextImageFrameToImageOverlay() {
+    // set image frame to image overlay
+    sceneView.imageOverlays[0].imageFrame = imageFrames[imageIndex]
+    // increment the index to keep track of which image to load next
+    imageIndex++
+    // reset index once all files have been loaded
+    if (imageIndex == imageFrames.size)
+      imageIndex = 0
+  }
+
+  /**
+   * Toggle's starting and stopping the timer on button tap.
```

Comment:
don't need an apostrophe here 
```suggestion
   * Toggles starting and stopping the timer on button tap.
```